system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp

```diff
@@ -1081,6 +1081,151 @@ bool AzureLogAnalyticsCurlClient::processSearchJsonResp(LogQueryResultDetails &
     return true;
 }
 
+bool AzureLogAnalyticsCurlClient::healthReport(StringBuffer & report, LogAccessHealthReportOptions options)
+{
+    try
+    {
+        report.appendf("\"ConnectionInfo\": { \"TargetALAWorkspaceID\": \"%s\" ", m_logAnalyticsWorkspaceID.str());
+        report.appendf(", \"TargetALATenantID\": \"%s\"", m_aadTenantID.str());
+        report.appendf(", \"TargetALAClientID\": \"%s\"", m_aadClientID.str());
+        report.appendf(", \"TargetALASecret\": \"%sempty\"", m_aadClientSecret.length()==0 ? "" : "not ");
+        report.appendf(", \"TargetsContainerLogV2\": \"%s\"", targetIsContainerLogV2 ? "true" : "false");
+        report.appendf(", \"ComponentsQueryJoins\": \"%sabled\"", m_disableComponentNameJoins ? "dis" : "en");
+        report.appendf(", \"BlobModeUnstructuredLogData\": \"%sabled\"", m_blobMode ? "en" : "dis");
+        report.append( "}"); //close conninfo
+
+        report.append(", \"ConfigurationInfo\": { ");
+
+        if (m_pluginCfg)
+        {
+            StringBuffer configJSON;
+            toJSON(m_pluginCfg, configJSON, 0);
+            report.appendf("\"ConfigurationTree\": %s", configJSON.str()); //json encode
+        }
```

Comment:
Is there any sensitive information in the configuration that shouldn't be exposed, like maybe tokens or passwords or internal URLs? On the other hand maybe this is a moot question since ECLWatch already gives access to component configs for roxie, esp etc.

Reply:
HELM chart Schema doesn't call for any sensitive data, I think we're ok.
https://github.com/hpcc-systems/HPCC-Platform/blob/70ccefc3d3a435a5596dc33f6d163d0f19b118c6/helm/hpcc/values.schema.json#L3192